### PR TITLE
Add check into CMake for bigendian platforms

### PR DESCRIPTION
**Pull request by axic**
And reject compilation if so.

I know this was for a later milestone, but wanted to get this clear.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,6 +19,12 @@ cable_set_build_type(DEFAULT Release CONFIGURATION_TYPES Debug;Release;Coverage)
 
 project(fizzy LANGUAGES CXX)
 
+include(TestBigEndian)
+TEST_BIG_ENDIAN(IS_BIG_ENDIAN)
```
Comment (chfast) [inline CMakeLists.txt]:
```suggestion
test_big_endian(is_big_endian)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -19,6 +19,12 @@ cable_set_build_type(DEFAULT Release CONFIGURATION_TYPES Debug;Release;Coverage)
...
+if (IS_BIG_ENDIAN)
```
Comment (chfast) [inline CMakeLists.txt]:
```suggestion
if(is_big_endian)
```

File: CMakeLists.txt (hunk 1 above)
Comment (axic) [reply CMakeLists.txt]: This is a copy paste from Solidity and it works (checked with `NOT IS_BIG_ENDIAN`).

What is the difference between lower and upper case?

Comment (chfast) [reply CMakeLists.txt]: No functional difference, just naming convention. CMake in general started using lower case for function names. Moreover, I prefer using lower case private variables.

Comment (axic) [reply CMakeLists.txt]: So are cmake variables/function names case insensitive or `TestBigEndian` exports both `test_big_endian` and `TEST_BIG_ENDIAN`?

Comment (chfast) [reply CMakeLists.txt]: It is case insensitive but only for function/macro names to my understanding. Variable names are case sensitive.

Comment (axic) [reply CMakeLists.txt]: Thanks for the clarification. Updated both.

File: CMakeLists.txt [hunk 3]
```diff
@@ -19,6 +19,12 @@ cable_set_build_type(DEFAULT Release CONFIGURATION_TYPES Debug;Release;Coverage)
 
 project(fizzy LANGUAGES CXX)
 
+include(TestBigEndian)
+test_big_endian(is_big_endian)
+if (is_big_endian)
```
Comment (chfast) [inline CMakeLists.txt]:
```suggestion
if(is_big_endian)
```

Comment (axic) [reply CMakeLists.txt]: Changed. I always wondered why there isn't a space there

File: CMakeLists.txt [hunk 4]
```diff
@@ -19,6 +19,12 @@ cable_set_build_type(DEFAULT Release CONFIGURATION_TYPES Debug;Release;Coverage)
 
 project(fizzy LANGUAGES CXX)
 
+include(TestBigEndian)
+test_big_endian(is_big_endian)
+if(is_big_endian)
+        message(FATAL_ERROR "${PROJECT_NAME} currently does not support big endian systems.")
```
Comment (chfast) [inline CMakeLists.txt]:
```suggestion
    message(FATAL_ERROR "${PROJECT_NAME} currently does not support big endian systems.")
```

File: CMakeLists.txt (hunk 3 above)
Comment (chfast) [reply CMakeLists.txt]: Because
```
if ()
else()
endif()
```

File: CMakeLists.txt (hunk 4 above)
Comment (axic) [reply CMakeLists.txt]: Can you add a cmake-format for cmakefiles? :)

Comment (chfast) [reply CMakeLists.txt]: It actually exists: https://github.com/cheshirekow/cmake_format
But my editor is good enough.

**Pull request merged**